File: workflows/app_test.py
"""
Simple Flask server workflow for GloriaMundo Test

This workflow runs the main Flask application for testing purposes.
"""

import os
import sys
import subprocess
import logging

# Configure logging
logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
    handlers=[logging.StreamHandler()]
)

logger = logging.getLogger(__name__)

def run():
    """
    Run the Flask application for testing
    """
    try:
        logger.info("Starting GloriaMundo App Test Workflow")
        logger.info("--------------------------------------")
        logger.info("This workflow will run the Flask application")
        logger.info("The application will be available at the webview URL")
        logger.info("Press Ctrl+C to stop the workflow")
        logger.info("--------------------------------------")
        
        # Run the Flask application
        subprocess.run(["python", "app.py"])
        
    except KeyboardInterrupt:
        logger.info("Workflow stopped by user")
    except Exception as e:
        logger.error(f"Error running workflow: {e}")
        sys.exit(1)

if __name__ == "__main__":
    run()

File: workflows/app_test.js
const { exec } = require('child_process');
const express = require('express');
const path = require('path');

// Print information message
console.log('Starting GloriaMundo Test Workflow');
console.log('--------------------------------------');
console.log('This workflow will run the Flask application');
console.log('The application will be available at the webview URL');
console.log('Press Ctrl+C to stop the workflow');
console.log('--------------------------------------');

// Execute the Flask application command
const flaskApp = exec('python app.py');

// Forward stdout and stderr to the console
flaskApp.stdout.on('data', (data) => {
  console.log(`${data}`);
});

flaskApp.stderr.on('data', (data) => {
  console.error(`${data}`);
});

// Handle process exit
flaskApp.on('close', (code) => {
  console.log(`Flask application exited with code ${code}`);
});

// Create a simple health check server
const app = express();
const PORT = 3000;

app.get('/', (req, res) => {
  res.send('GloriaMundo Test Workflow is running. The Flask app should be available at the webview URL.');
});

app.listen(PORT, () => {
  console.log(`Health check server listening on port ${PORT}`);
});